watchtower/wtclient/backup_task.go

```diff
@@ -0,0 +1,344 @@
+	// output to spend from.
+	if t.breachInfo.LocalOutputSignDesc != nil {
+		justiceKit.CommitToRemotePubKey = toBlobPubKey(
+			keyRing.NoDelayKey,
+		)
+	}
+
+	// Now, begin construction of the justice transaction. We'll start with
+	// a version 2 transaction.
+	justiceTxn := wire.NewMsgTx(2)
+
+	// Next, add the non-dust inputs that were derived from the breach
+	// information. This will either be contain both the to-local and
+	// to-remote outputs, or only be the to-local output.
+	inputs := t.inputs()
+	for prevOutPoint := range inputs {
+		justiceTxn.AddTxIn(&wire.TxIn{
+			PreviousOutPoint: prevOutPoint,
+		})
+	}
+
+	// Add the sweep output paying directly to the user, which was already
+	// verified as non-dust.
+	justiceTxn.AddTxOut(&wire.TxOut{
+		Value:    t.sweepAmt,
+		PkScript: sweepAddress,
+	})
+
+	// If the session is to have a reward output and one using the computed
+	// reward amount, which has already been checked to assert that it is
+	// not dusty.
+	if t.blobType.Has(blob.FlagReward) {
+		justiceTxn.AddTxOut(&wire.TxOut{
+			Value:    t.rewardAmt,
+			PkScript: t.rewardAddress,
+		})
+	}
+
+	// Sort the justice transaction according to BIP69.
+	txsort.InPlaceSort(justiceTxn)
+
+	// Check that the justice transaction meets basic validity requirements
+	// before attempting to attach the witnesses.
+	btx := btcutil.NewTx(justiceTxn)
+	if err := blockchain.CheckTransactionSanity(btx); err != nil {
+		return hint, nil, err
+	}
+
+	// Construct a sighash cache to improve signing performance.
+	hashCache := txscript.NewTxSigHashes(justiceTxn)
+
+	// Since the transaction inputs could have been reordered as a result of
+	// the BIP69 sort, create an index mapping each prevout to it's new
+	// index.
+	inputIndex := make(map[wire.OutPoint]int)
+	for i, txIn := range justiceTxn.TxIn {
+		inputIndex[txIn.PreviousOutPoint] = i
+	}
+
+	// Now, iterate through the list of inputs that were initially added to
+	// the transaction and store the computed witness within the justice
+	// kit.
+	for _, inp := range inputs {
+		// Lookup the input's new post-sort position.
+		i := inputIndex[*inp.OutPoint()]
+
+		// Construct the full witness required to spend this input.
+		inputScript, err := inp.CraftInputScript(
+			signer, justiceTxn, hashCache, i,
+		)
+		if err != nil {
+			return hint, nil, err
+		}
+
+		// Parse the DER-encoded signature from the first position of
+		// the resulting witness. We trim an extra byte to remove the
+		// sighash flag.
+		witness := inputScript.Witness
+		rawSignature := witness[0][:len(witness[0])-1]
+
+		// Reencode the DER signature into a fixed-size 64 byte
+		// signature.
+		signature, err := lnwire.NewSigFromRawSignature(rawSignature)
+		if err != nil {
+			return hint, nil, err
+		}
+
+		// Finally, copy the serialized signature into the justice kit,
+		// using the input's witness type to select the appropriate
+		// field.
+		switch inp.WitnessType() {
+		case input.CommitmentRevoke:
```

Comment:
Again, shouldn't these be flipped? We revoke their output, but then just sweep ours w/ no delay.